account_bank_statement_import_camt/models/account_bank_statement_import.py

```diff
@@ -33,7 +18,9 @@ def _parse_file(self, cr, uid, data_file, context=None):
         parser = self.pool['account.bank.statement.import.camt.parser']
         try:
             _logger.debug("Try parsing with camt.")
-            return parser.parse(data_file)
+            user_id = self.pool('res.users').browse(cr, uid, uid)
+            camt_import_batch = user_id.company_id.camt_import_batch
```

Comment:
@thomaspaulb thanks!
I think you can reduce a lot the code of your PR by avoid to pass `camt_import_batch` through the various methods.
Instead of reading `camt_import_batch = user_id.company_id.camt_import_batch` here and then passing `camt_import_batch` to the parser methods, you can just read the camt_import_batch configuration in the place where you need it. So the above two lines could be moved directly to `def parse_amount()` and `def parse_entry()`.
This way the signatures of the methods will remain untouched.

Reply:
That is true, but do you have any idea how could we get to `user_id` from the parser without refactoring the whole parser? There is not even `self.env` in the functions inside the parser.

Reply:
@thomaspaulb you're right! This part of code uses the old v7 api and the uid is not propagated through the parser. Let's keep the solution you adopted, it's ok as it is. Thanks!